liquibase-core/src/main/java/liquibase/integration/spring/SpringResource.java

```diff
@@ -31,7 +59,10 @@ public InputStream openInputStream() throws IOException {
     }
 
     @Override
-    public OutputStream openOutputStream() throws IOException {
+    public OutputStream openOutputStream(boolean createIfNeeded) throws IOException {
+        if (!resource.exists() && !createIfNeeded) {
+            throw new IOException("Resource "+getUri()+" does not exist");
+        }
```

Comment:
This logic should be abstracted up a layer, since this check will have to occur in every implementation of Resource.

Reply:
I'm not sure how it can, though? The `resource` variable there is a springframework.Resource class, so that is implementation-specific code.